packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -788,13 +788,38 @@ class _CupertinoBackGestureController<T> {
     }
 
     if (controller.isAnimating) {
+      late VoidCallback animationValueCallback;
+      late AnimationStatusListener animationStatusCallback;
+
+      // Keep the userGestureInProgress in true state so we don't change the
+      // curve of the page transition mid-flight since CupertinoPageTransition
+      // depends on userGestureInProgress. Switch the userGestureInProgress to
+      // false near the end, but just before so the app can be interacted with
+      // when the animation visibly ends.
+      animationValueCallback = () {
+        if (controller.value < 0.02) {
+          if (navigator.userGestureInProgress) {
+            navigator.didStopUserGesture();
+          }
+          controller.removeListener(animationValueCallback);
+          controller.removeStatusListener(animationStatusCallback);
+        }
+      };
+      // Only apply the value listener if going back to the previous route, as
+      // allowing clicks on the top route while in motion may trigger errors
+      // for going to a route that is already being moved to.
+      if (!animateForward) {
```

Comment:
It looks like based on whether we're animating forward, we need either the animationValueCallback or the animationStatusCallback, but never both?

Reply:
Ah nvm it looks like the route has access to the controller too.

Reply:
My worry is somehow the animation would avoid the animationValueCallback and the user would be locked out of gestures, especially because the animation duration can vary a lot. I was worried it would somehow avoid checking the 0.2 value. It was possible from testing locally if the value was extreme, like 0.00001, so I didn't want to risk it. That was my thought process, anyway.

Reply:
If it's unnecessary, I'm open to removing it though.